### PR TITLE
Shape optimization

### PR DESCRIPTION
**Pull request by MarkoLeskovar**
Dear Andreas,

This pull request is regarding the work discussed before our meeting today, I just wanted to finish old stuff so we don't have un-tracked progress further on. 

I short, I updated the GiD problem type so it should work with both Fluids and Structures and on top of that it is now clean of any unnecessary settings and therefore much easier to use. Furthermore I added one simple demonstration of different sampling methods and I also fixed a bug in random uniform distribution so it is consistent with other sampling methods. 

Let's close this pull request asap and I will start working on the first part of the contact problem.

Best regards,
Marko

### CODE REVIEW

**Review by andreas-apostolatos — CHANGES_REQUESTED** [review]
Hey Marko,

thanks for the changes. I tried to use the problem type of the structure for a simple case but when I try to write out the .dat file GiD14 crashes with the following error,

Error in base file '/home/andreas/software/GiD/gid14.01.1-x64/.problemtype/matlab.gid/matlab.bas' at line number 30: It's necessary a loop for 'materials'

Can you please figure it out? Otherwise I like very much the idea of having them split, thanks for the effort.

Greetings,
Andreas

**Comment by MarkoLeskovar** [comment]
Dear Andreas,

I fixed minor issues we discussed yesterday.

**Comment by andreas-apostolatos** [comment]
Hello Marko,

I ran a fluid case and the interface seems to work quite smoothly. However, we have to still address the following issues,

- Please enable the application of Dirichlet boundary conditions and ALE motion at nodes. This should overwrite conditions applied on intersecting lines.

- We need to be able to specify beginning, end time and either time step size or number of time steps already at GiD. For the moment we can specify if the analysis is steady-state or transient but we can specify neither beginning and end time nor the time discretization.

- Please address also the issue of having the nodes and elements assigned to the fluid domain already before meshing so that we avoid the redundant step after meshing.

Thank you very much for your efforts.

Best Regards,
Andreas

**Comment by MarkoLeskovar** [comment]
Dear Andreas,

I fixed the stuff you requested last time. Everything seems to be working. How I normally setup the problem is the following:

1. Draw geometry
2. Assign domains (nodes, elements)
3. Assign material to the surface
4. Assign the boundary conditions (watch out for the order in which you do it, lines still overwrite lines, I add "over points" as last step)
5. Assign ALE nodes (first I assign lines, then points)
6. Create mesh
7. Calculate

At ALE-nodes you can now specify lines and/or points and furthermore if it's a "free boundary". I kind of forgot which name to give it to be honest :) I did not changed the parser since I am not exactly sure how do you want to have it setup in MATLAB but basically there is a 1 or 0 at each ALE-node after the function handle so you know which nodes are on the boundary and which are inside.

Best regards,
Marko

File: MonteCarloSimulationAnalysis/probabilityDistributionFunctions/generateRandomUniformDistribution.m [hunk 1]
```diff
@@ -20,6 +20,9 @@
 %      distribution : Random vector of noSamples uniformly distributed
 %       
 %% Function main body
-distribution = 2*standardDeviation*rand(noSamples,1) + meanValue - standardDeviation;
+intervalSize = 0.5*standardDeviation*sqrt(12);
```
Comment (andreas-apostolatos) [inline MonteCarloSimulationAnalysis/probabilityDistributionFunctions/generateRandomUniformDistribution.m:23]: What is the reason behind these changes here?

File: problemTypeGiD/matlab.gid/matlab.bas [hunk 2]
```diff
@@ -35,28 +24,38 @@
...
 
 STRUCTURE_TRANSIENT_ANALYSIS
- SOLVER *GenData(STR_TIME_ANA-TYPE)
- TIME_INTEGRATION *GenData(STR_TIME_INTEGRATION-SCHEME)
- START_TIME *GenData(STR_START_TIME)
- END_TIME *GenData(STR_END_TIME)
- NUMBER_OF_TIME_STEPS *GenData(STR_NUMBER_OF_TIME_STEPS)
+ SOLVER *GenData(STR_Time_Analysis_Type)
+ TIME_INTEGRATION *GenData(STR_Time_Integration_Scheme)
+ ALPHA_BETA *GenData(STR_Alpha/Beta)
```
Comment (andreas-apostolatos) [inline problemTypeGiD/matlab.gid/matlab.bas]: Change variable STR_Alpha/Beta to STR_AlphaBeta the slash is dangerous

**Review by andreas-apostolatos — COMMENTED** [review]
Hello Marko,

thanks a lot for the changes, the new problem type works very nicely! Can you please correct the couple of minor issues I noted? Please pull before you make changes cause I already corrected some minor mistakes.

Greetings,
Andreas

File: MonteCarloSimulationAnalysis/probabilityDistributionFunctions/generateRandomUniformDistribution.m (hunk 1 above)
Comment (MarkoLeskovar) [reply MonteCarloSimulationAnalysis/probabilityDistributionFunctions/generateRandomUniformDistribution.m:23]: There was an inconsisteny in the formulas. All other distribution functions are defined to take in the standard deviation and mean value. With random uniform distribution however we had a mistake in our AFEM project code. 
If you put in for example meanValue = 10 and standardDeviation = 1 you would get a uniformly distributed values between 9 and 11. This makes sense, but that does not give back standard deviation of 1 as in the other cases. With the intervalSize we fix that problem as the interval size is calculated via standard deviation

**Comment by MarkoLeskovar** [comment]
Hello Andreas,

Could you check you latest changes / commits on this pull request, you said I should pull before I make changes, but I don't see any updates since 3 days ago.

Best regards,
Marko

**Comment by andreas-apostolatos** [comment]
> Hello Andreas,
> 
> Could you check you latest changes / commits on this pull request, you said I should pull before I make changes, but I don't see any updates since 3 days ago.
> 
> Best regards,
> Marko

Hey Marko,

I think you need to do git fetch or so. There needs to be definitely something new in this branch!

When could you drop by to discuss the further developments?

Greetings,
Andreas

**Comment by MarkoLeskovar** [comment]
> 
> 
> > Hello Andreas,
> > Could you check you latest changes / commits on this pull request, you said I should pull before I make changes, but I don't see any updates since 3 days ago.
> > Best regards,
> > Marko
> 
> Hey Marko,
> 
> I think you need to do git fetch or so. There needs to be definitely something new in this branch!
> 
> When could you drop by to discuss the further developments?
> 
> Greetings,
> Andreas

Dear Andreas,

I tried to fetch and don´t see anything, even in GitHub is does not show any changes...
I can drop by today at around 16:00 or tomorrow morning.

Best regards,
Marko

**Comment by andreas-apostolatos** [comment]
> > > Hello Andreas,
> > > Could you check you latest changes / commits on this pull request, you said I should pull before I make changes, but I don't see any updates since 3 days ago.
> > > Best regards,
> > > Marko
> > 
> > 
> > Hey Marko,
> > I think you need to do git fetch or so. There needs to be definitely something new in this branch!
> > When could you drop by to discuss the further developments?
> > Greetings,
> > Andreas
> 
> Dear Andreas,
> 
> I tried to fetch and don´t see anything, even in GitHub is does not show any changes...
> I can drop by today at around 16:00 or tomorrow morning.
> 
> Best regards,
> Marko

Hey Marko,

it would be great if you could drop by today.

Greetings,
Andreas
